homeassistant/helpers/entity.py

```diff
@@ -319,6 +322,53 @@ def unique_id(self) -> str | None:
         """Return a unique ID."""
         return self._attr_unique_id
 
+    @property
+    def use_device_name(self) -> bool:
+        """Return if this entity does not have its own name.
+
+        Should be True if the entity represents the single main feature of a device.
+        """
+
+        def report_implicit_device_name() -> None:
```

Comment:
Would we optimize performance by making a method of this function?

Reply:
we would gain a little bit, but I'd prefer to look into caching the name itself so it's not calculated every time the entity's name property is accessed.